common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
+                return true;
+            }
+
+            // If we can already execute, there is nothing to do.
+            if (file.canExecute()) {
+                return true;
+            }
+
+            // On volumes, with noexec set, even files with the executable POSIX permissions will fail to execute.
+            // So, check for executable permissions, then find out if the file can be executed.
+
+            // Everything is done reflectively to prevent the animal-sniffer plugin from complainin.
+            Class<?> filesClass = Class.forName("java.nio.file.Files");
```

Comment:
The important here is that you "MUST" import the java7 classes in another class that you only access if the version check match. Otherwise you will get a classloader exception when try to use java6